text2graph/schema.py

```diff
@@ -1,66 +1,116 @@
+    _class: str | None = None
+    color: str | None = None
+    fill: int | None = None
+    t_units: int | None = None
+
+    def hydrate(self) -> None:
+        """Hydrate Lithology from macrostrat."""
+        try:
+            hit = get_lith_records(self.name, exact=True)[0]
+        except (ValueError, IndexError):
+            logging.info(f"No records found for lithology '{self.name}'")
+            return
+
+        for k, v in hit.items():
+            if k == "class":
+                setattr(self, "_class", v)
+            else:
+                setattr(self, k, v)
+
 
 class Stratigraphy(BaseModel):
     strat_name: str
-    strat_name_long: str
-    rank: str
-    strat_name_id: int
-    concept_id: int
-    bed: str
-    bed_id: int
-    mbr: str
-    mbr_id: int
-    fm: str
-    fm_id: int
-    subgp: str
-    subgp_id: int
-    gp: str
-    gp_id: int
-    sgp: str
-    sgp_id: int
-    b_age: float
-    t_age: float
-    b_period: str
-    t_period: str
-    c_interval: str
-    t_units: int
-    ref_id: int
+    strat_name_long: str | None = None
+    rank: str | None = None
+    strat_name_id: int | None = None
+    concept_id: int | None = None
+    bed: str | None = None
+    bed_id: int | None = None
+    mbr: str | None = None
+    mbr_id: int | None = None
+    fm: str | None = None
+    fm_id: int | None = None
+    subgp: str | None = None
+    subgp_id: int | None = None
+    gp: str | None = None
+    gp_id: int | None = None
+    sgp: str | None = None
+    sgp_id: int | None = None
+    b_age: float | None = None
+    t_age: float | None = None
+    b_period: str | None = None
+    t_period: str | None = None
+    c_interval: str | None = None
+    t_units: int | None = None
+    ref_id: int | None = None
+
+    def model_post_init(self, __context) -> None:
+        self.hydrate()
+
+    def hydrate(self) -> None:
+        """Hydrate Stratigraphy from macrostrat."""
+        try:
+            hit = get_strat_records(self.strat_name, exact=True)[0]
+        except (ValueError, IndexError):
+            logging.info(f"No records found for stratigraphy '{self.strat_name}'")
+            return
+
+        for k, v in hit.items():
+            setattr(self, k, v)
+
 
 def valid_longitude(v: float) -> float:
-    assert -180 <= v <= 180, f'{v} is not a valid longitude'
+    assert -180 <= v <= 180, f"{v} is not a valid longitude"
     return v
 
+
 def valid_latitude(v: float) -> float:
-    assert -90 <= v <= 90, f'{v} is not a valid latitude'
+    assert -90 <= v <= 90, f"{v} is not a valid latitude"
     return v
 
-Latitude = Annotated[float, AfterValidator(valid_latitude)]
-Longitude = Annotated[float, AfterValidator(valid_longitude)]
 
 class Location(BaseModel):
     name: str
-    lat: Latitude | None
-    lon: Longitude | None
+    lat: Annotated[float, AfterValidator(valid_latitude)] | None = None
+    lon: Annotated[float, AfterValidator(valid_longitude)] | None = None
+
+    def hydrate(self) -> None:
```

Comment:
@ilmcconnell 
The `hydrate` function will try to get the lat and long from some geocoding services.  Can you help with this?

Reply:
@JasonLo I adapted the geolocation code from SERPAPI and threw it in here - we'll need to be careful with the number of locations declared, since it will go through the free API calls pretty quickly.